### DISCUSSION TITLE
course/starbook/register-overview

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#72] [General]
# course/starbook/register-overview

星辰英语本项目的用户注册功能的开发，详细阐述了GoFrame框架的代码分层设计和ORM功能的应用，提供了MySQL驱动的安装与使用说明，通过简单示例说明数据库交互的实现过程，提高代码可读性与维护性，避免SQL注入风险，并与MySQL数据库的实际连接过程进行了讲解。

https://goframe.org/course/starbook/register-overview

**Comment by yinglunfeng** [comment]
这里controller调用logic没有经过service层似乎更简单，有点困惑service层到底用于什么场景。

**Reply by oldme-git** [reply]
`service` 只是 `logic` 生成的接口，完全忽略`service`，`controller` 直接到 `Logic` 也是可以的

**Reply by yinglunfeng** [reply]
给常感谢回答。我知道service层主要是接口抽象，框架设计里面也是建议controller和logic之间通过service解耦，但是实际写起来，直接controller调用logic更简单，service层再定义接口似乎多了一些冗余代码，所以想了解service层什么场景会更有存在意义

**Reply by oldme-git** [reply]
没有什么场景更有存在意义，它就是个抽象出来的接口

**Reply by sjg20010414** [reply]
刚好今天学习重新理解。个人理解，如果每一个controller恰好只使用一个logic对象，那么service几乎是没有意义的，因为此时controller直接持有logic对象的指针就行了。但如果controller ctrlA 用到了 logic对象 lgA, lgC，controller ctrlB 用到了 logic对象 lgB lgC，那么，加入service层，用服务查找来获取业务逻辑对象lgC的指针，可以避免重新创建对象——这或许在大型系统中能感受到差别，我不是专职程序员，应该感受不到这种CPU和内存的差别。

**Comment by huaiqianqian** [comment]
panic: configuration file "config.toml" not found, did you miss the configuration file or the misspell the configuration file name? 
当我按照教程运行时 发生这样的错误 但我将./manifest/config/config.yaml 移动到 ./config.yaml 问题解决了 我觉得我严格遵循教程了

**Reply by Xing-Fax** [reply]
我也遇到了此问题
将 "github.com/gogf/gf/frame/g"更改为"github.com/gogf/gf/v2/frame/g"解决

**Comment by yinglunfeng** [comment]
亲，您的邮件我已收到，我看到后会尽快恢复您的！

**Comment by tonyho** [comment]
最后一步执行前，需要手动创建star数据库，否则出现无法连接的错误

**Comment by yinglunfeng** [comment]
亲，您的邮件我已收到，我看到后会尽快恢复您的！

**Comment by LogicalMonster** [comment]
更新gf到v2.9.0后安装github.com/gogf/gf/contrib/drivers/mysql/v2，运行gf up出现github.com/olekukonko/tablewriter v1.0.9版本冲突问题，手动指定github.com/olekukonko/tablewriter v0.0.5解决冲突。

**Reply by LogicalMonster** [reply]
测试后发现使用go get -u github.com/gogf/gf/contrib/drivers/mysql/v2安装会有版本冲突，使用go get github.com/gogf/gf/contrib/drivers/mysql/v2安装则没有版本冲突

**Comment by yinglunfeng** [comment]
亲，您的邮件我已收到，我看到后会尽快恢复您的！

**Comment by yinglunfeng** [comment]
亲，您的邮件我已收到，我看到后会尽快恢复您的！
